contracts/FastBreak.cdc

```diff
@@ -0,0 +1,708 @@
+        pub var winner: Address? /// The address of the winner of Fast Break
+        pub var submissions: {Address: FastBreak.FastBreakSubmission} /// Map of each submission to the Fast break
+        pub let fastBreakRunID: String /// The off-chain uuid of the Fast Break Run containing this Fast Break
+        pub var stats: [FastBreakStat] /// The NBA statistical requirements for this Fast Break
+
+        init (
+            id: String,
+            name: String,
+            fastBreakRunID: String,
+            submissionDeadline: UInt64,
+            numPlayers: UInt64
+        ) {
+            if let fb = FastBreak.fastBreakGameByID[id] {
+                self.id = fb.id
+                self.name = fb.name
+                self.submissionDeadline = fb.submissionDeadline
+                self.numPlayers = fb.numPlayers
+                self.status = fb.status
+                self.winner = fb.winner
+                self.submissions = fb.submissions
+                self.fastBreakRunID = fb.fastBreakRunID
+                self.stats = fb.stats
+            } else {
+                self.id = id
+                self.name = name
+                self.submissionDeadline = submissionDeadline
+                self.numPlayers = numPlayers
+                self.status = FastBreak.GameStatus.SCHEDULED
+                self.submissions = {}
+                self.fastBreakRunID = fastBreakRunID
+                self.stats = []
+                self.winner = nil
+            }
+        }
+
+        /// Get a account's active Fast Break Submission
+        ///
+        pub fun getFastBreakSubmissionByAccount(accountAddress: Address): FastBreak.FastBreakSubmission? {
+            let fastBreakSubmissions = self.submissions!
+
+            return fastBreakSubmissions[accountAddress]
+        }
+
+        /// Add a statistic to the Fast Break during game creation
+        ///
+        access(contract) fun addStat(stat: FastBreak.FastBreakStat) {
+            self.stats.append(stat)
+        }
+
+        /// Update status and winner of a Fast Break
+        ///
+        access(contract) fun update(status: FastBreak.GameStatus, winner: Address?) {
+            self.status = status
+            self.winner = winner
+        }
+
+        /// Submit a Fast Break
+        ///
+        access(contract) fun submitFastBreak(submission: FastBreak.FastBreakSubmission) {
+            pre {
+                FastBreak.isValidSubmission(submissionDeadline: self.submissionDeadline) : "submission missed deadline"
+            }
+
+            self.submissions[submission.accountAddress] = submission
+        }
+
+        /// Update the Fast Break score of an account
+        ///
+        access(contract) fun updateScore(accountAddress: Address, points: UInt64, win: Bool): Bool {
+            var isNewWin = false
+            let submissions = self.submissions
+            let submission: FastBreak.FastBreakSubmission = submissions[accountAddress]!
```

Comment:
Remember to use `if...let` instead of force unwrap

Reply:
I think I got all of these, thanks Josh.